library/core/src/slice/memchr.rs

```diff
@@ -105,6 +105,8 @@ pub fn memrchr(x: u8, text: &[u8]) -> Option<usize> {
     let (min_aligned_offset, max_aligned_offset) = {
         // We call this just to obtain the length of the prefix and suffix.
         // In the middle we always process two chunks at once.
+        // SAFETY: transmuting `[u8]` to `[usize]` is safe except for size differences
+        // which are handled by `align_to`.
```

Comment:
That's only safe if the `[u8]` is correctly aligned (and has a size that holds an integer multiple of `usize`s as you write here AFAIU). Might be worth mentioning that `align_to` also takes care of aligning properly (does it?).

Reply:
The more precise safety condition would be that `[u8; size_of::<usize>]` has the same validity invariants as usize, but I would not block on that in this PR.